sympy/physics/control/lti.py

```diff
@@ -1422,3 +1424,326 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+    mutate your original ``TransferFunctionMatrix``.
+
+    >>> from sympy import I
+    >>> tfm_3.subs({s: I})  #  substituting imaginary i (iota) to s
+    Matrix([
+    [-5*I, 5*I],
+    [-5*I, 5/1]])
+
+    See Also
+    ========
+
+    TransferFunction, Series, Parallel, Feedback
+
+    """
+    def __new__(cls, arg):
+
+        expr_mat_arg = []
+        var = arg[0][0].var
+        for row in range(len(arg)):
+            temp = []
+            for col in range(len(arg[row])):
+                if not isinstance(arg[row][col], (TransferFunction, Series, Parallel)):
+                    raise TypeError("Incompatible type found as the element of TransferFunctionMatrix")
+
+                if var != arg[row][col].var:
+                    raise ValueError("Conflicting value(s) found for `var`. All TransferFunction instances in "
+                            "TransferFunctionMatrix should use the same complex variable in Laplace domain.")
+
+                if isinstance(arg[row][col], TransferFunction):
+                    temp.append(arg[row][col].to_expr())
+                else:
+                    temp.append(arg[row][col].doit().to_expr())
+            expr_mat_arg.append(temp)
+
+
+        if isinstance(arg, (list, Tuple)):
+            # Making nested Tuple (sympy.core.containers.Tuple) from nested list or nested python tuple
+            arg = Tuple(*(Tuple(*r, sympify=False) for r in arg), sympify=False)
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
+        obj._expr_mat = ImmutableMatrix(expr_mat_arg)
+        obj._var = arg[0][0].var
+        obj._num_outputs, obj._num_inputs = obj._expr_mat.shape
+        return obj
+
+    @property
+    def var(self):
+        """
+        Returns the complex variable used by all the transfer functions or
+        ``Series``/``Parallel`` objects in a transfer function matrix.
+
+        If explicitly provided as the third argument of ``TransferFunctionMatrix``, then returns
+        the same, else, ``TransferFunctionMatrix`` automatically looks up the ``var`` used by
+        transfer functions inside the list/tuple provided as the first argument and returns that.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import p, s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix, Series, Parallel
+        >>> G1 = TransferFunction(p**2 + 2*p + 4, p - 6, p)
+        >>> G2 = TransferFunction(p, 4 - p, p)
+        >>> G3 = TransferFunction(0, p**4 - 1, p)
+        >>> G4 = TransferFunction(s + 1, s**2 + s + 1, s)
+        >>> S1 = Series(G1, G2)
+        >>> S2 = Series(-G3, Parallel(G2, -G1))
+        >>> tfm1 = TransferFunctionMatrix([[G1], [G2], [G3]])
+        >>> tfm1.var
+        p
+        >>> tfm2 = TransferFunctionMatrix([[-S1, -S2], [S1, S2]])
+        >>> tfm2.var
+        p
+        >>> tfm3 = TransferFunctionMatrix([[G4]])
```

Comment:
Extra square brackets look redundant. Do we really have to restrict `arg` to be nested list? Can this work like Matrix?
```
In [28]: Matrix([TransferFunction(s**2, s - 1, s).to_expr()])          
Out[28]: Matrix([[s**2/(s - 1)]])
```

Reply:
I first planned to do it like this but that would have required another conditional expression in the constructor. Since I wanted to object creation to be as fast and efficient as possible, I thought that would be better to put it as a compulsory condition for the user.

This was there in my implementation in PR #21556. 
https://github.com/sympy/sympy/blob/f6bd2f18eff977ea094a6b9606f352ac1224337a/sympy/physics/control/lti.py#L1981-L1983

I also thought that it would be better for a beginner to have a simple API (PEP 20). If we allow normal lists then we would have to further explain that it is supposed to be a column matrix and not row. For new users who are not accustomed to sympy matrices might find it difficult to adapt. 

I would do whatever your suggest but allowing simple lists as `arg` might be a performance tradeoff.

Reply:
OK, let's go with this.